core/src/storage2/collections/binary_heap/mod.rs

```diff
@@ -269,3 +275,162 @@ where
         value
     }
 }
+
+/// An iterator over shared references to the elements of the `BinaryHeap`.
+#[derive(Debug, Clone, Copy)]
+pub struct Iter<'a, T>
+where
+    T: PackedLayout + Ord,
+{
+    /// The storage vector to iterate over.
+    elems: &'a StorageVec<Children<T>>,
+    /// The current begin of the iteration.
+    begin: u32,
+    /// The current end of the iteration.
+    end: u32,
```

Comment:
Might as well internally simply use `StorageVec`'s iterator. Should also be more efficient probably. I haven't checked this though, so please take my comment with a grain of salt.